src/query/sql/src/planner/binder/table.rs

```diff
@@ -260,10 +260,13 @@ impl Binder {
                     }
                     parent = bind_context.parent.as_mut();
                 }
-                return Err(ErrorCode::UnknownTable(format!(
-                    "Unknown table `{database}`.`{table_name}` in catalog '{catalog}'"
-                ))
-                .set_span(*span));
+                if e.code() == 1025 {
```

Comment:
From @drmingdrmer https://github.com/datafuselabs/databend/pull/14076#pullrequestreview-1790063645 : 

Use named const ErrorCode::UNKNOWN_TABLE instead of integer.

Does it need to deal with other causing errors such as UnknownDatabase? There are several other errors needs to be considered: WrongShare, WrongShareObject, UnknownTableId